params/precompile_upgrade.go

```diff
@@ -149,18 +148,33 @@ func (c *ChainConfig) verifyPrecompileUpgrades() error {
 	return nil
 }
 
-// GetActivePrecompileConfig returns the most recent precompile config corresponding to [address].
+// verifyStateUpgrades checks [c.StateUpgrades] is well formed:
+// - the specified blockTimestamps must monotonically increase
+func (c *ChainConfig) verifyStateUpgrades() error {
+	var previousUpgradeTimestamp *big.Int
+	for i, upgrade := range c.StateUpgrades {
+		upgradeTimestamp := upgrade.BlockTimestamp
+		// Verify specified timestamps are strictly monotonically increasing.
```

Comment:
should we check prohibited addreses here in case we don't override any of them (especially precompiles)? I'd say if we want this to be a `alloc`-like mechanism in genesis, then we can keep these pretty much limitless. but still might worth considering to limit those.

Reply:
I think we can leave this unchecked and document this as an emergency developer tool.